.gitlab-ci.yml

```diff
@@ -490,118 +487,40 @@ build-rust-doc:
-    - if: $CI_COMMIT_REF_NAME == "tags"
-    - if: $CI_COMMIT_REF_NAME =~ /^v[0-9]+\.[0-9]+.*$/              # i.e. v1.0, v2.1rc1
-    - if: $CI_COMMIT_MESSAGE =~ /\[chaos:(basic|medium|large)\]/ && $CI_COMMIT_REF_NAME =~ /^[0-9]+$/  # i.e add [chaos:basic] in commit message to trigger
-  image:                           paritypr/simnet:latest
-  needs:
-    - job:                         build-chaos-docker
-  tags:
-    - parity-simnet
-  variables:
-    <<:                            *default-vars
-    PRODUCT:                       substrate
-    DOCKERFILE:                    $PRODUCT.Dockerfile
-    CONTAINER_IMAGE:               paritypr/$PRODUCT
-    KEEP_NAMESPACE:                0
-    NAMESPACE:                     "substrate-ci-${CI_COMMIT_SHORT_SHA}-${CI_PIPELINE_ID}"
-    VERSION:                       "ci-${CI_COMMIT_SHORT_SHA}"
-  interruptible: true
-  environment:
-    name:                          parity-simnet
-  script:
-    - simnet spawn dev -i $CONTAINER_IMAGE:$VERSION
-    - simnet singlenodeheight -h 30
-  after_script:
-    - simnet clean
-
 #### stage:                        publish
 
 .build-push-docker-image:          &build-push-docker-image
   <<:                              *build-refs
   <<:                              *kubernetes-build
-  image:                           docker:stable
-  services:
-    - docker:dind
+  image:                           quay.io/buildah/stable
   variables:                       &docker-build-vars
     <<:                            *default-vars
-    DOCKER_HOST:                   tcp://localhost:2375
-    DOCKER_DRIVER:                 overlay2
     GIT_STRATEGY:                  none
     DOCKERFILE:                    $PRODUCT.Dockerfile
-    CONTAINER_IMAGE:               parity/$PRODUCT
+    IMAGE_NAME:                    docker.io/parity/$PRODUCT
   before_script:
-    - test "$Docker_Hub_User_Parity" -a "$Docker_Hub_Pass_Parity"
-        || ( echo "no docker credentials provided"; exit 1 )
-    - docker login -u "$Docker_Hub_User_Parity" -p "$Docker_Hub_Pass_Parity"
-    - docker info
+    - test "$Docker_Hub_User_Parity" -a "$Docker_Hub_Pass_Parity" ||
+        ( echo "no docker credentials provided"; exit 1 )
   script:
     - cd ./artifacts/$PRODUCT/
     - VERSION="$(cat ./VERSION)"
     - echo "${PRODUCT} version = ${VERSION}"
     - test -z "${VERSION}" && exit 1
-    - docker build
+    - buildah bud
+      --squash
+      --format=docker
       --build-arg VCS_REF="${CI_COMMIT_SHA}"
       --build-arg BUILD_DATE="$(date -u '+%Y-%m-%dT%H:%M:%SZ')"
-      --tag $CONTAINER_IMAGE:$VERSION
-      --tag $CONTAINER_IMAGE:latest
-      --file $DOCKERFILE .
-    - docker push $CONTAINER_IMAGE:$VERSION
-    - docker push $CONTAINER_IMAGE:latest
+      --tag "$IMAGE_NAME:$VERSION"
+      --tag "$IMAGE_NAME:latest"
+      --file "$DOCKERFILE" .
+    - echo "$Docker_Hub_Pass_Parity" |
+      buildah login --username "$Docker_Hub_User_Parity" --password-stdin docker.io
```

Comment:
Are we no longer using ALL_CAPS format for secrets in jobs?

Reply:
this is how it was saved historically in GitLab, will be changed with moving to secrets storage